code/components/gta-streaming-five/src/GameEventHooks.cpp

```diff
@@ -62,18 +107,144 @@ void* HandleEventWrap(void* group, rage::fwEvent* event)
 		}
 		catch (std::exception& e)
 		{
-		
 		}
 	}
 
 	return (*TFunc)(group, event);
 }
 
-#include <sstream>
+template<decltype(g_eventCall1)* TFunc>
+void* HandleEventWrapReact(rage::fwEventGroup* group, rage::fwEvent* event)
+{
+	if (event)
+	{
+		/*
+		 * event:
+		 * These events are stored with different member variables and with different layouts per class/struct,
+		 * so we have to investigate them case by case.
+		 * 
+		 * group:
+		 * In this reacting version the entity that triggers this event is stored in group->entities[17]
+		 * as well as in group->entities[18].
+		 */
 
-namespace rage
+		try
+		{
+			msgpack::sbuffer buf;
+			msgpack::packer packer(buf);
+
+			const char* eventName = typeid(*event).name() + 6;
+
+			packer.pack_array(4); // we'll offer 4 parameters
+			msgpack_pack(packer, eventName, event->GetId(), GetGuidFromBaseSafe(group->entities[17]));
+
+			// Quick 4 bytes check, i.e.: no strcmp
+			// this will turn the first 4 bytes after "CEvent" into an uint32 and check against that value
+			switch ((uint32_t&)eventName[6])
```

Comment:
Most code in this project makes intent clear by using `*(T*)`, `(T&)` is needlessly confusing and different.

Reply:
What do you prefer instead? reinterpret_casts?

Reply:
Again, `*(T*)` is the canonical choice elsewhere in the project.

Reply:
Ah the comma there made me think I couldn't use both, bit confusing.